hydra_server/server/complexmodels.py

```diff
@@ -1496,6 +1498,16 @@ def __init__(self, parent=None):
         self.status      = parent.status
         self.cr_date     = str(parent.cr_date)
         self.created_by  = parent.created_by
+
+        #this is a failsafe to ensure that the 'appdata' column
+        #is processed correctly, as it is a JSON column type which may
+        #not work with with serialising data.
+        appdata = {}
+        if parent.appdata is not None:
+            for k, v in parent.appdata.items():
+                appdata[k] = v
+        self.appdata = appdata
```

Comment:
This creates a copy of `parent.appdata` and then sets `self.appdata` to it, so is equivalent to...

```python
    if parent.appdata is not None:
        self.appdata = parent.appdata.copy()
```

Is `parent` guaranteed to have an `appdata` attr here?